http.go

```diff
@@ -108,34 +108,37 @@ func (g *Gateway) GraphQLHandler(w http.ResponseWriter, r *http.Request) {
 		operations = append(operations, operation)
 		// or we got a POST request
 	} else if r.Method == http.MethodPost {
-		// read the full request body
-		body, err := ioutil.ReadAll(r.Body)
-		if err != nil {
-			payloadErr = fmt.Errorf("encountered error reading body: %s", err.Error())
-		}
+		body, fileMap, extractError := extractBody(r)
+		if extractError != nil {
+			payloadErr = extractError
+		} else {
+			// there are two possible options for receiving information from a post request
+			// the first is that the user provides an object in the form of { query, variables, operationName }
+			// the second option is a list of that object
+
+			singleQuery := &HTTPOperation{}
+			// if we were given a single object
+			if err := json.Unmarshal(body, &singleQuery); err == nil {
+				// add it to the list of operations
+				operations = append(operations, singleQuery)
+				// we weren't given an object
+			} else {
+				// but we could have been given a list
+				batch := []*HTTPOperation{}
 
-		// there are two possible options for receiving information from a post request
-		// the first is that the user provides an object in the form of { query, variables, operationName }
-		// the second option is a list of that object
+				if err = json.Unmarshal(body, &batch); err != nil {
+					payloadErr = fmt.Errorf("encountered error parsing body: %s", err.Error())
+				} else {
+					operations = batch
+				}
 
-		singleQuery := &HTTPOperation{}
-		// if we were given a single object
-		if err = json.Unmarshal(body, &singleQuery); err == nil {
-			// add it to the list of operations
-			operations = append(operations, singleQuery)
-			// we weren't given an object
-		} else {
-			// but we could have been given a list
-			batch := []*HTTPOperation{}
+				// we're in batch mode
+				batchMode = true
+			}
 
-			if err = json.Unmarshal(body, &batch); err != nil {
+			if err := injectFiles(operations, fileMap, batchMode); err != nil {
```

Comment:
I'm not sure im following what is happening at this step. could you add a comment, please?

Reply:
This functions associates files in upload request with respective variables (initially they are not associated according to the spec the variables are just set to null. Maybe that's should be moved inside of `extract` (or shall we name it `parseRequest`) method?